### PR TITLE
[11.0][MIG] base_transaction_id

### PR DESCRIPTION
**Pull request by yvaucher**

### CODE REVIEW

File: base_transaction_id/README.rst
Comment (ghost) [inline base_transaction_id/README.rst]: is that email correct?

File: base_transaction_id/README.rst
Comment (ghost) [inline base_transaction_id/README.rst]: From latest [README.rst template](https://github.com/OCA/maintainer-tools/blame/master/template/module/README.rst#L55):
`help us smash it by providing detailed and welcomed feedback.`

File: base_transaction_id/i18n/base_transaction_id.pot
Comment (ghost) [inline base_transaction_id/i18n/base_transaction_id.pot]: Remove this file

File: base_transaction_id/__manifest__.py
Comment (ghost) [inline base_transaction_id/__manifest__.py]: https://github.com/OCA/account-reconcile

File: base_transaction_id/__manifest__.py
Comment (ghost) [inline base_transaction_id/__manifest__.py]: Separate authors w/ space, please

File: base_transaction_id/views/invoice.xml
Comment (ghost) [inline base_transaction_id/views/invoice.xml:4]: `id` must be declared before `model`

**Review by ghost — CHANGES_REQUESTED** [review]

File: base_transaction_id/README.rst
Comment (ghost) [inline base_transaction_id/README.rst]: https://github.com/OCA/account-reconcile/issues

**Review by ghost — CHANGES_REQUESTED** [review]

File: base_transaction_id/views/invoice.xml
Comment (yvaucher) [reply base_transaction_id/views/invoice.xml:4]: Please give me a reference on this, according to the OCA codebase both are used:

└─> ack record.*model=.*id= --xml | wc -l
24252
└─> ack record.*id=.*model= --xml | wc -l
21737


EDIT: wrong greping
EDIT2: removing occurence of OpenUpgrade which is not revelent

File: base_transaction_id/__manifest__.py
Comment (yvaucher) [reply base_transaction_id/__manifest__.py]: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#modules

According to contribution guidelines this is correct 

``` python
author + ',Odoo Community Association (OCA)
```

File: base_transaction_id/README.rst
Comment (yvaucher) [reply base_transaction_id/README.rst]: I removed it as he didn't contributed directly to that module, he has few commits through OCA module maintenance

**Comment by yvaucher** [comment]
Thanks for the review, I did the fixes accordingly

File: base_transaction_id/README.rst
Comment (pedrobaeza) [reply base_transaction_id/README.rst]: Yeah, I haven't made anything significant here.

File: base_transaction_id/__manifest__.py
Comment (pedrobaeza) [reply base_transaction_id/__manifest__.py]: Odoo parses both with or without spaces, so no problem in any

File: base_transaction_id/README.rst
Comment (simahawk) [inline base_transaction_id/README.rst]: keep `ID` always uppercase, just to make it homogeneous when referring to it

File: base_transaction_id/__manifest__.py
Comment (simahawk) [inline base_transaction_id/__manifest__.py:1]: 2012-2018 ? Same for other headings

File: base_transaction_id/models/account_bank_statement_line.py
Comment (simahawk) [inline base_transaction_id/models/account_bank_statement_line.py]: self.ensure_one() ?

File: base_transaction_id/models/account_bank_statement_line.py
Comment (simahawk) [inline base_transaction_id/models/account_bank_statement_line.py]: not a requirement but a good habit: no space around docstring text + end w/ period ;)

File: base_transaction_id/models/account_move.py
Comment (simahawk) [inline base_transaction_id/models/account_move.py]: docstring: same

File: base_transaction_id/views/invoice.xml
Comment (simahawk) [inline base_transaction_id/views/invoice.xml]: I'd replace `group/group` with `//` to make it more robust in case is moved. Even if as you are selecting the field origin.... why not selecting only that? Very likely we don't have the same filed twice. No?

**Review by simahawk — CHANGES_REQUESTED** [review]
@yvaucher code LGTM. Just some nitpicking non-blocking remarks

File: base_transaction_id/__manifest__.py
Comment (yvaucher) [reply base_transaction_id/__manifest__.py:1]: https://github.com/OCA/account-reconcile/pull/190/commits/18615b14d20c81bd58a02681713aee58adb7caff

Here are the changes I made, I don't think it deserve a bump on copyright as no code was change in python files.

File: base_transaction_id/models/account_bank_statement_line.py
Comment (yvaucher) [reply base_transaction_id/models/account_bank_statement_line.py]: This was written at a time we considered ensure_one to be called anyway. I can add it thought

File: base_transaction_id/models/account_move.py
Comment (yvaucher) [inline base_transaction_id/models/account_move.py]: To change: using builtin `str`is a very bad practice.

Comment (yvaucher) [inline base_transaction_id/models/account_move.py:37]: Bad idea to use str as it is a `builtin` not sure were the super is defined though

Comment (yvaucher) [reply base_transaction_id/models/account_move.py:37]: Defined in src/addons/account/models/account_move.py

Comment (yvaucher) [reply base_transaction_id/models/account_move.py:37]: Let's try a PR on master https://github.com/odoo/odoo/pull/22392

File: base_transaction_id/views/invoice.xml
Comment (yvaucher) [reply base_transaction_id/views/invoice.xml]: Yes, can obviously be simplified.

It was probably failing in v10 as it was changed from
`<field name="origin" position="after">` to this.

If there are 1 origin field in the invoice lines before the origin field of the main object.

So it might be safer to stick with the expr but simplify it a bit.

**Review by simahawk — APPROVED** [review]

File: base_transaction_id/__manifest__.py
Comment (simahawk) [reply base_transaction_id/__manifest__.py]: @yvaucher remember to update this

**Comment by yvaucher** [comment]
Squashed commits

File: base_transaction_id/views/invoice.xml
Comment (grindtildeath) [reply base_transaction_id/views/invoice.xml:4]: @yvaucher https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#format

File: base_transaction_id/__manifest__.py
Comment (ghost) [reply base_transaction_id/__manifest__.py]: @pedrobaeza that just doesn't look good when it gets rendered, imho

File: base_transaction_id/models/account_bank_statement_line.py
Comment (ghost) [inline base_transaction_id/models/account_bank_statement_line.py]: `ensure_one`?

File: base_transaction_id/models/account_move.py
Comment (ghost) [inline base_transaction_id/models/account_move.py:17]: It's better to open the line right after the parenthesis :)

File: base_transaction_id/__manifest__.py
Comment (pedrobaeza) [reply base_transaction_id/__manifest__.py]: The parsing is the same as said. You can try.

File: base_transaction_id/models/invoice.py
Comment (ghost) [inline base_transaction_id/models/invoice.py:10]: same remark about code formatting

**Review by ghost — APPROVED** [review]
LGTM w/ some remarks

**Review by grindtildeath — APPROVED** [review]
Please update website key in __manifest__, otherwise LGTM :+1:

File: base_transaction_id/models/invoice.py
Comment (yvaucher) [reply base_transaction_id/models/invoice.py:10]: While I also prefer this format with return after opening parethesis I prefer not to change this to not alter too much the code base history.

File: base_transaction_id/models/account_bank_statement_line.py
Comment (yvaucher) [reply base_transaction_id/models/account_bank_statement_line.py]: Would make it more explicit right.

**Comment by yvaucher** [comment]
Changed website key in `__manifest__.py` and added an ensure_one.

Other demands seems irrelevant to me for that module migration.

**Comment by asaunier** [comment]
@yvaucher The PR has been approved and it seems there is nothing left to do with it. What about merging? :)

**Comment by yvaucher** [comment]
@OCA/accounting-maintainers anyone who could merge this?

As it is my PR I don't want to force merge.

**Comment by pedrobaeza** [comment]
Can you maybe squash a bit the commit history? :pray:

**Review by asaunier — APPROVED** [review]
Squashing some commits would indeed help clarifying the history but I have the feeling this PR could be merged as is. @yvaucher already did some squashing 2.5 months ago. Isn't this PR been opened for too long now? :P

**Comment by pedrobaeza** [comment]
Yeah, but 50 commits for such a simple module I think it's too much, but I let you the final call.

**Comment by yvaucher** [comment]
@pedrobaeza ok I'll squash few of my commits and drop the ones "Make modules uninstallable"

I guess I can easily make it around 40 commits, this is a quite old module thus we have many commits.

**Comment by yvaucher** [comment]
@pedrobaeza I squashed the commits in a new branch, thus I'm closing this one

https://github.com/OCA/account-reconcile/pull/206

**Pull request closed without merging**
